src/game/entities/creatures.ts

```diff
@@ -110,17 +96,35 @@ export default class Creature {
     pos: Position;
     template: CreatureTemplate;
   }) {
-    this.name = template.name as CreatureName;
+    const moveSpeed = ~~RNG(20, 600);
```

Comment:
I would strongly suggest changing to use `Math.floor`, it is more readable, more accurate (`~~` truncates, not floors, so -ve floats end up being rounded away from zero), and sometimes faster [jsperf](https://jsperf.com/jsfvsbitnot)

Reply:
Yeah I've been caught out by the truncation rather than flooring in some of the world stuff. I think I only did this because it was shorter syntax and it feels a bit cooler 😄 
I agree it's probably best to change, although looking at the jsperf results it looks like there was a time when bitwise NOT was more performant.

Reply:
It used to be faster, but has always been incorrect. With js, the performance benefit is probably negligible, but readibility and correctness is not.